### PR TITLE
5223 add permission check for update status

### LINKED ISSUE #5223: App crashes when trying to scan without permission
## What went wrong? 😲
Found while testing [THIS](https://github.com/msupply-foundation/open-msupply/issues/1912)

If there's no permission and we try to scan QR from the equipment tab in Android, the app crashes.
Screenshot 2024-10-24 at 4 37 27 pm

https://github.com/user-attachments/assets/02444d74-f731-40cc-af7b-4cfe7ad283e9

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to user permission and remove permission from the Permissions3> Assets
2. Now sync with the tablet and try to add equipment from scanner button
3. See error

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:  `V2.3.01-RC1`

- Platform: Android (V2.3.01-RC1)

- Database type:

### PR DESCRIPTION
**Pull request by aimee-mcneil-melville**
Fixes #5223

# 👩🏻‍💻 What does this PR do?

This is a sidequest off 5223 (PR #5839), as there was another button for Assets (Update Status) that also had no checks for permissions.

Adds permission checks to the Update Status button in Cold Chain -> Equipment -> Asset detail view, and notifies the user if they are not permitted

There are two pathways:
- [ ] if the user has 'view' permissions but not 'add/edit', when clicking the button they will notified they do not have permission to edit an asset

- [ ] if the user has 'add/edit' permissions, they will be able to select the update status button and sucessfully change the status of an asset

If the user has no 'view' permission for assets, they will not be able to access the area this button is in

The user will benefit from feedback on why they cannot access this function. This also prevents any further code being actioned if the permissions are not valid

## 💌 Any notes for the reviewer?

# 🧪 Testing

Can be tested on browser and/or tablet - see #5839

With all permissions on
- [ ]  Go to Cold Chain -> Equipment
- [ ]  Scan an existing asset QR code or click on one to open it
- [ ]  Click on the ```Update Status``` button and select a different status
- [ ] Click OK -> See that the status has been updated for the asset

With view only permissions on
- [ ] Have user permission on for View Asset and off for Add/Edit Asset in 4D -> sync -> go to equipment page
- [ ] Scan an existing asset QR code or click on one to open it
- [ ] Click on the ```Update Status``` button
- [ ] See a notification "You do not have permission to edit an asset"

No testing for 'no permissions'
As this button is accessed by selecting an asset, a user with no permission to view will not be able to get to this button

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/coldchain/src/Equipment/DetailView/UpdateStatusButton.tsx [hunk 1]
```diff
@@ -43,10 +45,19 @@ export const UpdateStatusButtonComponent = ({
...
+  const { error, success, info } = useNotification();
   const [draft, setDraft] = useState<Partial<Draft>>(getEmptyAssetLog(''));
   const { insertLog, invalidateQueries } = useAssets.log.insert();
 
+  const permission = UserPermission.AssetMutate;
+  const { userHasPermission } = useAuthContext();
+
+  const onUpdateStatus = () => {
+    if (userHasPermission(permission)) {
+      showDialog;
```
Comment (lache-melvin) [inline client/packages/coldchain/src/Equipment/DetailView/UpdateStatusButton.tsx:57]: Need to call this - the modal isn't opening 😁

**Review by lache-melvin — CHANGES_REQUESTED** [review]
Just need to fix the showDialog 😅 

Looks like a lot of the changes from your other branch are in here too? I tried changing the base of this PR, so that it is going into your other branch, but I can still see those changes - I think need to merge that branch into this one? Or merge this other one, then pull develop into this one, either should work 🙏

File: client/packages/coldchain/src/Equipment/DetailView/UpdateStatusButton.tsx (hunk 1 above)
Comment (aimee-mcneil-melville) [reply client/packages/coldchain/src/Equipment/DetailView/UpdateStatusButton.tsx]: Done, thanks!

**Comment by aimee-mcneil-melville** [comment]
> Just need to fix the showDialog 😅
> 
> Looks like a lot of the changes from your other branch are in here too? I tried changing the base of this PR, so that it is going into your other branch, but I can still see those changes - I think need to merge that branch into this one? Or merge this other one, then pull develop into this one, either should work 🙏

Yes, I should have noted it in here but this was branched off of the scan button branch.. It does look like now this will be going into the other branch?

**Review by lache-melvin — APPROVED** [review]
🔥 

Yeah looks great! Both are approved now so feel free to merge in either order!

**Pull request merged**